Makefile

```diff
@@ -59,163 +34,230 @@ ifeq ("$(DEBUG_BUILD)", "true")
+	@git -C '$(PGMONITOR_DIR)' checkout '$(PGMONITOR_VERSION)'
+	@git -C '$(PGMONITOR_DIR)' config pull.ff only
+
+.PHONY: get-postgres-exporter
+get-postgres-exporter:
+	[ ! -e hack/tools/postgres_exporter.tar.gz ] || (rm hack/tools/postgres_exporter.tar.gz && echo "Deleting old exporter")
+	wget -O hack/tools/postgres_exporter.tar.gz '$(POSTGRES_EXPORTER_URL)'
+
+.PHONY: clean
+clean: ## Clean resources
+clean: clean-deprecated
+	rm -f bin/postgres-operator
+	rm -f config/rbac/role.yaml
+	[ ! -d testing/kuttl/e2e-generated ] || rm -r testing/kuttl/e2e-generated
+	[ ! -d testing/kuttl/e2e-generated-other ] || rm -r testing/kuttl/e2e-generated-other
+	rm -rf build/crd/generated build/crd/*/generated
+	[ ! -f hack/tools/setup-envtest ] || hack/tools/setup-envtest --bin-dir=hack/tools/envtest cleanup
+	[ ! -f hack/tools/setup-envtest ] || rm hack/tools/setup-envtest
+	[ ! -d hack/tools/envtest ] || rm -r hack/tools/envtest
+	[ ! -d hack/tools/pgmonitor ] || rm -rf hack/tools/pgmonitor
+	[ ! -n "$$(ls hack/tools)" ] || rm -r hack/tools/*
+	[ ! -d hack/.kube ] || rm -r hack/.kube
 
-setup:
-	PGOROOT='$(PGOROOT)' ./bin/get-deps.sh
-	./bin/check-deps.sh
+.PHONY: clean-deprecated
+clean-deprecated: ## Clean deprecated resources
+	@# packages used to be downloaded into the vendor directory
+	[ ! -d vendor ] || rm -r vendor
+	@# executables used to be compiled into the $GOBIN directory
+	[ ! -n '$(GOBIN)' ] || rm -f $(GOBIN)/postgres-operator $(GOBIN)/apiserver $(GOBIN)/*pgo
+	@# executables used to be in subdirectories
+	[ ! -d bin/pgo-rmdata ] || rm -r bin/pgo-rmdata
+	[ ! -d bin/pgo-backrest ] || rm -r bin/pgo-backrest
+	[ ! -d bin/pgo-scheduler ] || rm -r bin/pgo-scheduler
+	[ ! -d bin/postgres-operator ] || rm -r bin/postgres-operator
+	@# keys used to be generated before install
+	[ ! -d conf/pgo-backrest-repo ] || rm -r conf/pgo-backrest-repo
+	[ ! -d conf/postgres-operator ] || rm -r conf/postgres-operator
 
-#=== postgrescluster CRD ===
+##@ Deployment
+.PHONY: createnamespaces
+createnamespaces: ## Create operator and target namespaces
+	kubectl apply -k ./config/namespace
 
-# Create operator and target namespaces
-createnamespaces:
-	$(PGO_KUBE_CLIENT) apply -k ./config/namespace
+.PHONY: deletenamespaces
+deletenamespaces: ## Delete operator and target namespaces
+	kubectl delete -k ./config/namespace
 
-# Delete operator and target namespaces
-deletenamespaces:
-	$(PGO_KUBE_CLIENT) delete -k ./config/namespace
+.PHONY: install
+install: ## Install the postgrescluster CRD
+	kubectl apply --server-side -k ./config/crd
 
-# Install the postgrescluster CRD
-install:
-	$(PGO_KUBE_CLIENT) apply --server-side -k ./config/crd
+.PHONY: uninstall
+uninstall: ## Delete the postgrescluster CRD
+	kubectl delete -k ./config/crd
 
-# Delete the postgrescluster CRD
-uninstall:
-	$(PGO_KUBE_CLIENT) delete -k ./config/crd
+.PHONY: deploy
+deploy: ## Deploy the PostgreSQL Operator (enables the postgrescluster controller)
```

Comment:
For `deploy` and `install`, would it make sense for `createnamespaces` to be a prerequisite?

Reply:
I was considering deleting the `createnamespace` stuff entirely. Does anyone use it? It's easy enough to create a namespace. I typically do it manually.

I think no. If we want a namespace to be a part of the installation process, we should put it in the kustomize :thinking: